pkg/ast/common.go

```diff
@@ -244,3 +245,30 @@ func ParseTimeRange(startEpoch, endEpoch uint64, aggs *QueryAggregators, qid uin
 	tRange.EndEpochMs = endEpoch
 	return tRange, nil
 }
+
+func GetDefaultTimechartSpanOptions(startEpoch, endEpoch uint64, qid uint64) (*structs.SpanOptions, error) {
+	if startEpoch == 0 || endEpoch == 0 {
+		err := fmt.Errorf("GetDefaultTimechartSpanOptions: , time range not set")
+		return nil, err
+	}
+
+	duration := endEpoch - startEpoch
+
+	// 15 minutes
+	if duration <= 15*60*1000 {
+		return &structs.SpanOptions{SpanLength: &structs.SpanLength{Num: 10, TimeScalr: utils.TMSecond}}, nil
+	} else if duration <= 60*60*1000 {
+		return &structs.SpanOptions{SpanLength: &structs.SpanLength{Num: 1, TimeScalr: utils.TMMinute}}, nil
+	} else if duration <= 4*60*60*1000 {
+		return &structs.SpanOptions{SpanLength: &structs.SpanLength{Num: 5, TimeScalr: utils.TMMinute}}, nil
+	} else if duration <= 24*60*60*1000 {
+		return &structs.SpanOptions{SpanLength: &structs.SpanLength{Num: 30, TimeScalr: utils.TMMinute}}, nil
+	} else if duration <= 7*24*60*60*1000 {
+		return &structs.SpanOptions{SpanLength: &structs.SpanLength{Num: 1, TimeScalr: utils.TMHour}}, nil
+	} else if duration <= 180*24*60*60*1000 {
+		return &structs.SpanOptions{SpanLength: &structs.SpanLength{Num: 1, TimeScalr: utils.TMDay}}, nil
```

Comment:
Where are you getting these values from? I was checking https://docs.splunk.com/Documentation/Splunk/9.1.1/SearchReference/Timechart#Default_time_spans and all the other `(duration, spanLength)` pairs match with what you have, but these two don't. Was there other documentation where you found these values?

Reply:
@AndrewHess No, I added them because, as with the Splunk documentation

- If it is **more than 30 days**, it will default to only 1 month. And since we have **90 days** in our time range, which is three months. However, I also felt just three data points would be very low. So I set it for 180 days, which is up to 6 months, the data will be given per day. Then after that, it will be given per month as how Splunk does.
- And regarding days, we have **2 days**, as one of the options, if it is more than a day, by default Splunk would choose per day interval; which will give us two data points. Thinking in a similar way as above, I set the interval as one hour for the range of more than a day and less than or equal **to 7 days**.

Reply:
Ok, that makes sense. What you have is fine then